### PR TITLE
feat: T-18441 User Permissions to cache/router

### PR DESCRIPTION
**Pull request by commoddity**
This PR adds User Permissions to cache and router of PHD.

Can be merged only once https://github.com/pokt-foundation/db-client/pull/11 is published and the E2E tests updated to include this PHD client version.

### CODE REVIEW

File: v1_cache/cache.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ type Cache struct {
 	blockchains              []*types.Blockchain
 	loadBalancersMap         map[string]*types.LoadBalancer
 	loadBalancersMapByUserID map[string][]*types.LoadBalancer
+	userPermissions          map[types.UserID]types.UserPermissions
```
Comment (pablocampogo) [inline v1_cache/cache.go:24]: This map should be to pointer

**Review by pablocampogo — CHANGES_REQUESTED** [review]
Also shouldn't an update to the listener be done?

**Comment by commoddity** [comment]
> 

It does update the listener but it's handled in the `add/update/deleteUserAccess` cache methods since User Access listener notifications are what trigger changes to the permissions struct.

**Review by pablocampogo — APPROVED** [review]

File: v1_cache/cache.go (hunk 1 above)
Comment (commoddity) [reply v1_cache/cache.go]: Fixed

**Pull request merged**